indigo_app/static/javascript/indigo/views/document_xml_editor.js

```diff
@@ -1,128 +1,336 @@
-(function(exports) {
-  "use strict";
+/**
+ * This class handles the text-based AKN editor. It is responsible for unparsing XML into text, re-parsing changed
+ * text into XML, and handling text-based editor actions (like bolding, etc.).
+ */
+class AknTextEditor {
+  constructor (root, document, onElementParsed) {
+    this.root = root;
+    this.document = document;
+    this.previousText = null;
+    this.xmlElement = null;
+    this.onElementParsed = onElementParsed;
+    // flag to prevent circular updates to the text
+    this.updating = false;
+    this.liveUpdates = false;
 
-  if (!exports.Indigo) exports.Indigo = {};
-  Indigo = exports.Indigo;
+    this.grammarName = document.tradition().settings.grammar.name;
+    this.grammarModel = new Indigo.grammars.registry[this.grammarName](
+      document.get('frbr_uri'),
+      document.url() + '/static/xsl/text.xsl');
```

Comment:
I think `/text.xsl` is going to keep returning null?

Reply:
yes, removed